### PR TITLE
Use Order CRUD to save admin order data

### PR DESCRIPTION
**Pull request by claudiosanches**
@mikejolley just not sure if we should keep salving the meta data from custom billing and shipping fields, since we support save it by default today.

### CODE REVIEW

File: includes/admin/meta-boxes/class-wc-meta-box-order-data.php [hunk 1]
```diff
@@ -482,49 +485,28 @@ public static function save( $post_id, $post ) {
...
-					'%s',
-				),
-				array(
-					'%d',
-				)
-			);
-		}
+		$props['date_created'] = $date;
+
+		clean_post_cache( $order_id );
```
Comment (mikejolley) [inline includes/admin/meta-boxes/class-wc-meta-box-order-data.php]: If this is needed, it should move to the CRUD class save method if not there already.

File: includes/admin/meta-boxes/class-wc-meta-box-order-data.php [hunk 2]
```diff
@@ -482,49 +485,28 @@ public static function save( $post_id, $post ) {
...
+		// Order data saved, now get it so we can manipulate status.
+		$order->set_props( $props );
+		$order->update();
 
-		// Order status
+		// Order status.
 		$order->update_status( $_POST['order_status'], '', true );
 
-		wc_delete_shop_order_transients( $post_id );
+		wc_delete_shop_order_transients( $order_id );
```
Comment (mikejolley) [inline includes/admin/meta-boxes/class-wc-meta-box-order-data.php]: Move to CRUD class.

File: includes/admin/meta-boxes/class-wc-meta-box-order-data.php [hunk 3]
```diff
@@ -482,49 +485,28 @@ public static function save( $post_id, $post ) {
...
 
-		// Order data saved, now get it so we can manipulate status
-		$order = wc_get_order( $post_id );
+		// Order data saved, now get it so we can manipulate status.
+		$order->set_props( $props );
+		$order->update();
 
-		// Order status
+		// Order status.
 		$order->update_status( $_POST['order_status'], '', true );
```
Comment (mikejolley) [inline includes/admin/meta-boxes/class-wc-meta-box-order-data.php]: Does this need to happen separate?

File: includes/admin/meta-boxes/class-wc-meta-box-order-data.php [hunk 4]
```diff
@@ -482,49 +485,28 @@ public static function save( $post_id, $post ) {
...
-		}
+		$props['date_created'] = $date;
+
+		clean_post_cache( $order_id );
 
-		// Order data saved, now get it so we can manipulate status
-		$order = wc_get_order( $post_id );
+		// Order data saved, now get it so we can manipulate status.
+		$order->set_props( $props );
+		$order->update();
```
Comment (mikejolley) [inline includes/admin/meta-boxes/class-wc-meta-box-order-data.php]: Use save() so create/update is handled automatically.

**Review by mikejolley — CHANGES_REQUESTED** [review]

**Comment by claudiosanches** [comment]
@mikejolley all updated.

**Pull request merged**